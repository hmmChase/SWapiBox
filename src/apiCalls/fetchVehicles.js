import { doFetch } from './doFetch';

export const fetchVehicles = async () => {
  const root = 'https://swapi.co/api';
  const vehiclesData = await doFetch(`${root}/vehicles`);
  return vehiclesData.results.map(vehicle => {
    return {
      name: vehicle.name,
      model: vehicle.model,
      class: vehicle.vehicle_class,
      passengers: vehicle.passengers
    };
  });
}
