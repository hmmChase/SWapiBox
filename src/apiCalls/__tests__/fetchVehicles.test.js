import { fetchVehicles } from "../fetchVehicles";

describe('fetchVehicles', () => {

  it('fetchVehicles is defined', () => {
    expect(fetchVehicles).toBeDefined();
  });
  
});