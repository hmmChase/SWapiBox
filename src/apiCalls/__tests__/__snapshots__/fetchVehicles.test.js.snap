// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchVehicles matches snapshot 1`] = `[Function]`;
